### PR TITLE
SSH connection plugin: add changelog for migration to config system (#73708)

### PR DESCRIPTION
**Pull request by pilou-**
##### SUMMARY
SSH connection plugin: add a changelog fragment for migration to config system (#73708)

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
doc

##### ADDITIONAL INFORMATION
Thanks to @Glandos for pointing that.

### CODE REVIEW

File: changelogs/fragments/73708-ssh-plugin-migrated-to-config-system.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+breaking_changes:
+  - "ssh connection - SSH connection plugin uses the config system, some parameters have been renamed and moved from the global configuration to the SSH connection plugin. For example use ``lookup('config', 'control_path', plugin_name='ssh', plugin_type='connection')`` instead of ``lookup('config', 'ANSIBLE_SSH_CONTROL_PATH')``. The affected parameters are: ``ssh_args`` (was ``ANSIBLE_SSH_ARGS``), ``control_path`` (was ``ANSIBLE_SSH_CONTROL_PATH``), ``control_path_dir`` (was ``ANSIBLE_SSH_CONTROL_PATH_DIR``), ``ssh_executable`` (was ``ANSIBLE_SSH_EXECUTABLE``), ``reconnection_retries`` (was ``ANSIBLE_SSH_RETRIES``), ``scp_if_ssh`` (was ``DEFAULT_SCP_IF_SSH``), ``sftp_batch_mode`` (was ``DEFAULT_SFTP_BATCH_MODE``), ``ssh_transfer_method`` (was ``DEFAULT_SSH_TRANSFER_METHOD``)."
```
Comment (bcoca) [inline changelogs/fragments/73708-ssh-plugin-migrated-to-config-system.yml:2]: The changes are not in substitution, they were meant to be additive (the source of configuration in code non withstanding), the ways the user could change the setting should still be the same with a few additions.

Any change that resulted in substitution should be corrected.

Comment (bcoca) [inline changelogs/fragments/73708-ssh-plugin-migrated-to-config-system.yml:2]: The changes are not in substitution, they were meant to be additive (the source of configuration in code non withstanding), the ways the user could change the setting should still be the same with a few additions.

Any change that resulted in substitution should be corrected.

Comment (bcoca) [inline changelogs/fragments/73708-ssh-plugin-migrated-to-config-system.yml:2]: The changes are not in substitution, they were meant to be additive (the source of configuration in code non withstanding), the ways the user could change the setting should still be the same with a few additions.

Any change that resulted in substitution should be corrected.

Comment (bcoca) [inline changelogs/fragments/73708-ssh-plugin-migrated-to-config-system.yml:2]: The changes are not in substitution, they were meant to be additive (the source of configuration in code non withstanding), the ways the user could change the setting should still be the same with a few additions.

Any change that resulted in substitution should be corrected.

Comment (bcoca) [inline changelogs/fragments/73708-ssh-plugin-migrated-to-config-system.yml:2]: The changes are not in substitution, they were meant to be additive (the source of configuration in code non withstanding), the ways the user could change the setting should still be the same with a few additions.

Any change that resulted in substitution should be corrected.

Comment (bcoca) [inline changelogs/fragments/73708-ssh-plugin-migrated-to-config-system.yml:2]: The changes are not in substitution, they were meant to be additive (the source of configuration in code non withstanding), the ways the user could change the setting should still be the same with a few additions.

Any change that resulted in substitution should be corrected.

Comment (bcoca) [inline changelogs/fragments/73708-ssh-plugin-migrated-to-config-system.yml:2]: The changes are not in substitution, they were meant to be additive (the source of configuration in code non withstanding), the ways the user could change the setting should still be the same with a few additions.

Any change that resulted in substitution should be corrected.

Comment (bcoca) [inline changelogs/fragments/73708-ssh-plugin-migrated-to-config-system.yml:2]: The changes are not in substitution, they were meant to be additive (the source of configuration in code non withstanding), the ways the user could change the setting should still be the same with a few additions.

Any change that resulted in substitution should be corrected.

Comment (bcoca) [inline changelogs/fragments/73708-ssh-plugin-migrated-to-config-system.yml:2]: The changes are not in substitution, they were meant to be additive (the source of configuration in code non withstanding), the ways the user could change the setting should still be the same with a few additions.

Any change that resulted in substitution should be corrected.

Comment (pilou-) [reply changelogs/fragments/73708-ssh-plugin-migrated-to-config-system.yml:2]: I tested the described behavior with these commands:
```
$ cat ansible.cfg
[ssh_connection]
control_path = /tmp/test
$ ansible --version
ansible 2.10.17
  config file = /home/pilou/work/ansible.cfg
  configured module search path = ['/home/pilou/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/pilou/.venv/ansible-4.10/lib/python3.9/site-packages/ansible
  executable location = /home/pilou/.venv/ansible-4.10/bin/ansible
  python version = 3.9.10 (main, Jan 16 2022, 17:12:18) [GCC 11.2.0]
$ ansible all -ilocalhost,  -mdebug -amsg='{{ lookup("config", "ANSIBLE_SSH_CONTROL_PATH") }}'
localhost | SUCCESS => {
    "msg": "/tmp/toto"
}

$ # switch to ansible 2.12

$ ansible --version
ansible [core 2.12.1]
  config file = /home/pilou/work/ansible.cfg
  configured module search path = ['/home/pilou/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/pilou/.venv/ansible5/lib/python3.9/site-packages/ansible
  ansible collection location = /home/pilou/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/pilou/.venv/ansible5/bin/ansible
  python version = 3.9.10 (main, Jan 16 2022, 17:12:18) [GCC 11.2.0]
  jinja version = 3.0.3
  libyaml = True
$ ansible all -ilocalhost,  -mdebug -amsg='{{ lookup("config", "ANSIBLE_SSH_CONTROL_PATH") }}'
localhost | FAILED! => {
    "msg": "Unable to find setting ANSIBLE_SSH_CONTROL_PATH. module 'ansible.constants' has no attribute 'ANSIBLE_SSH_CONTROL_PATH'. module 'ansible.constants' has no attribute 'ANSIBLE_SSH_CONTROL_PATH'"
}
```

Comment (bcoca) [reply changelogs/fragments/73708-ssh-plugin-migrated-to-config-system.yml:2]: i don't doubt it, i have some PRs fixing some of these issues, the point is that it was not intended as a breaking change and any breaking change is a bug, not something to be documented.

The 'old config variables' should have stayed for a deprecation period.

Comment (pilou-) [reply changelogs/fragments/73708-ssh-plugin-migrated-to-config-system.yml:2]: Is this a critical issue for 2.11? https://docs.ansible.com/ansible/devel/reference_appendices/release_and_maintenance.html#ansible-core-changelogs states that 2.11 releases will contain only critical bug fixes.

Comment (bcoca) [reply changelogs/fragments/73708-ssh-plugin-migrated-to-config-system.yml:2]: i would say so, but not my call

Comment (pilou-) [reply changelogs/fragments/73708-ssh-plugin-migrated-to-config-system.yml:2]: I tested with ansible-core 2.11.10 and 2.12.4, the changes that resulted in substitution are still here:
```
$ ansible all -ilocalhost,  -mdebug -amsg='{{ lookup("config", "ANSIBLE_SSH_CONTROL_PATH") }}'
localhost | FAILED! => {
    "msg": "Unable to find setting ANSIBLE_SSH_CONTROL_PATH. module 'ansible.constants' has no attribute 'ANSIBLE_SSH_CONTROL_PATH'. module 'ansible.constants' has no attribute 'ANSIBLE_SSH_CONTROL_PATH'"
```

**Comment by sivel** [comment]
I'm going to go ahead and close this as 2.11 is only receiving security fixes.  We had marked this PR as being on-hold a few months ago, because we weren't agreed on this change, and that status has not changed.

**Pull request closed without merging**